homeassistant/components/backup/config.py

```diff
@@ -2,56 +2,116 @@
 
 from __future__ import annotations
 
-from dataclasses import dataclass
+from dataclasses import dataclass, field, replace
 from datetime import datetime
-from typing import Self, TypedDict
+from typing import TYPE_CHECKING, Self, TypedDict
 
-from homeassistant.core import HomeAssistant
+from cronsim import CronSim
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.event import async_track_point_in_time
 from homeassistant.helpers.storage import Store
 from homeassistant.helpers.typing import UNDEFINED, UndefinedType
+from homeassistant.util import dt as dt_util
 
 from .const import DOMAIN
 
+if TYPE_CHECKING:
+    from .manager import BackupManager
+
+CRON_PATTERN_DAILY = "0 2 * * *"
+CRON_PATTERN_WEEKLY = "0 2 * * {}"
+DELAY_SAVE = 30
 STORAGE_KEY = DOMAIN
 STORAGE_VERSION = 1
 
 
 class StoredBackupConfig(TypedDict):
     """Represent the stored backup config."""
 
+    addons_included: list[str] | None
+    agent_ids: list[str]
+    database_included: bool
+    folders_included: list[str] | None
     last_automatic_backup: datetime | None
     max_copies: int | None
+    name: str | None
+    password: str | None
+    schedule: StoredBackupSchedule
+
+
+class StoredBackupSchedule(TypedDict):
+    """Represent the stored backup schedule."""
+
+    daily: bool
+    never: bool
+    weekday: str | None
 
 
 @dataclass(kw_only=True)
 class BackupConfigData:
     """Represent loaded backup config data."""
 
+    addons_included: list[str] | None = None
+    agent_ids: list[str] = field(default_factory=list)
+    database_included: bool = True
+    folders_included: list[str] | None = None
     last_automatic_backup: datetime | None = None
     max_copies: int | None = None
+    name: str | None = None
+    password: str | None = None
+    schedule: BackupSchedule
 
     @classmethod
     def from_dict(cls, data: StoredBackupConfig) -> Self:
         """Initialize backup config data from a dict."""
+        schedule_data = data["schedule"]
+        schedule = BackupSchedule(
+            daily=schedule_data["daily"],
+            never=schedule_data["never"],
+            weekday=schedule_data["weekday"],
+        )
         return cls(
+            addons_included=data["addons_included"],
+            agent_ids=data["agent_ids"],
+            database_included=data["database_included"],
+            folders_included=data["folders_included"],
             last_automatic_backup=data["last_automatic_backup"],
             max_copies=data["max_copies"],
+            name=data["name"],
+            password=data["password"],
+            schedule=schedule,
         )
 
     def to_dict(self) -> StoredBackupConfig:
```

Comment:
Why do we need this method?

Reply:
The store only allows a mapping as generic type.